plugins/FakeCardReader.cpp

```diff
@@ -72,8 +87,15 @@ FakeCardReader::do_conf(const data_t& args)
     cfg_ = args.get<fakecardreader::Conf>();
 
     // Read input
-    source_buffer_ = std::make_unique<FileSourceBuffer>(cfg_.input_limit, constant::WIB_SUPERCHUNK_SIZE);
-    source_buffer_->read(cfg_.data_filename);
+    if (cfg_.raw_type == "wib") {
+      source_buffer_ = std::make_unique<FileSourceBuffer>(cfg_.input_limit, constant::WIB_SUPERCHUNK_SIZE);
+      source_buffer_->read(cfg_.data_filename);
+    }
+
+    if (cfg_.raw_type == "tp") {
```

Comment:
We didn't want to have either WIB or TP as raw type, BUT extending the configuration by ENABLING the TP link.
(e.g.: cfg_.tps_enabled ). Please extend the schema for the FakeCardReader, so this configuration is available. Then check inside the cfg_.raw_type == WIB statement, if with tps or not.